runtime-sdk/src/dispatcher.rs

```diff
@@ -107,11 +117,19 @@ impl<R: Runtime> Dispatcher<R> {
             return e.to_call_result();
         }
 
-        match R::Modules::dispatch_call(ctx, &call.method, call.body) {
-            module::DispatchResult::Handled(result) => result,
-            module::DispatchResult::Unhandled(_) => {
-                modules::core::Error::InvalidMethod.to_call_result()
+        match call.method.as_str() {
+            // Internal methods.
+            BATCH_WEIGHT_LIMIT_QUERY_METHOD => {
+                let block_weight_limits = R::Modules::get_block_weight_limits(ctx);
+                CallResult::Ok(cbor::to_value(block_weight_limits))
             }
```

Comment:
Hm this doesn't seem right, it should be down below in the `query` handler?

Reply:
fixed